src/hashdict.jl

```diff
@@ -525,10 +525,19 @@ end
 
 start(t::HashDict) = skip_deleted(t, 1)
 done(t::HashDict, i) = done(t.vals, i)
-next(t::HashDict, i) = ((t.keys[i],t.vals[i]), skip_deleted(t,i+1))
+if VERSION >= v"0.4.0-dev+980"
+    next(t::HashDict, i) = (Pair(t.keys[i],t.vals[i]), skip_deleted(t,i+1))
+else
+    next(t::HashDict, i) = ((t.keys[i],t.vals[i]), skip_deleted(t,i+1))
+end
+
 
 done{K,V}(t::HashDict{K,V,Ordered}, i) = done(t.order, i)
-next{K,V}(t::HashDict{K,V,Ordered}, i) = ((t.keys[t.order[i]],t.vals[t.order[i]]), skip_deleted(t,i+1))
+if VERSION >= v"0.4.0-dev+980"
+    next{K,V}(t::HashDict{K,V,Ordered}, i) = (Pair(t.keys[t.order[i]],t.vals[t.order[i]]), skip_deleted(t,i+1))
+else
+    next{K,V}(t::HashDict{K,V,Ordered}, i) = ((t.keys[t.order[i]],t.vals[t.order[i]]), skip_deleted(t,i+1))
+end
```

Comment:
Base didn't actually start returning Pairs until pretty recently, even though the change to using Pairs everywhere else happened at `v"0.4.0-dev+980"`.  

This is probably okay, though--I doubt people are depending on this, and I doubt this would break code even if they were.

(Comments to the contrary welcome.)